### PR TITLE
Add "how it works" section in readme

### PR DESCRIPTION
**Pull request by majk-p**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
 
-[![License](http://img.shields.io/:license-Apache%202-green.svg)](http://www.apache.org/licenses/LICENSE-2.0.txt)
+[![License](https://img.shields.io/:license-Apache%202-green.svg)](http://www.apache.org/licenses/LICENSE-2.0.txt)
 ![Continuous Integration](https://github.com/pitgull/pitgull/workflows/Continuous%20Integration/badge.svg)
 [![Powered by cats](https://img.shields.io/badge/powered%20by-cats-blue.svg)](https://github.com/typelevel/cats)
 ![Gluten free](https://img.shields.io/badge/gluten-free-orange.svg)
 
+## How it works
+
+The core idea behind the project is very simple - create a tool for automatically applying merge requests created by [Scala Steward](https://github.com/scala-steward-org/scala-steward) on any [Gitlab](https://gitlab.com/) instance.
```
Comment (kubukoz) [inline README.md:10]: That was the core idea, but the app's goal is to be orthogonal from the bots making the PRs

File: README.md [hunk 2]
```diff
@@ -1,10 +1,19 @@
...
+
+Pitgull works as a web service, listening for webhook events from Gitlab instance. Upon receiving a webhook, it reads related project's merge requests, finds one best candidate and tries to merge it. Once it's merged, Gitlab would trigger another webhook, creating a loop until Pitgull finds no qualifying merge request.
```
Comment (kubukoz) [inline README.md:12]:
```suggestion
Pitgull works as a web service, listening for webhook events from Gitlab instance. Upon receiving a webhook, it reads related project's merge requests, finds one best candidate and tries to merge it. If all the qualifying merge requests need to be updated, an attempt to rebase them onto the target branch will be made.
Once the MR merged, Gitlab would trigger another webhook, creating a loop until Pitgull finds no qualifying merge request.
```

File: README.md (hunk 1 above)
Comment (majk-p) [reply README.md]: Fair point, added a sentence about this.

**Review by kubukoz — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
 ![Continuous Integration](https://github.com/pitgull/pitgull/workflows/Continuous%20Integration/badge.svg)
 [![Powered by cats](https://img.shields.io/badge/powered%20by-cats-blue.svg)](https://github.com/typelevel/cats)
 ![Gluten free](https://img.shields.io/badge/gluten-free-orange.svg)
 
+## How it works
+
+The core idea behind the project is very simple - create a tool for automatically applying merge requests created by [Scala Steward](https://github.com/scala-steward-org/scala-steward) on any [Gitlab](https://gitlab.com/) instance. Currently Pitgull is ortogonal from the bots, the merge requests are qualified by generic rules.
+
+Pitgull works as a web service, listening for webhook events from Gitlab instance. Upon receiving a webhook, it reads related project's merge requests, finds one best candidate and tries to merge it. If all the qualifying merge requests need to be updated, an attempt to rebase them onto the target branch will be made.
+Once the MR merged, Gitlab would trigger another webhook, creating a loop until Pitgull finds no qualifying merge request.
```
Comment (pkowalcze) [inline README.md:13]:
```suggestion
Once an MR is merged, Gitlab will trigger another webhook, creating a loop until Pitgull finds no qualifying merge request.
```

File: README.md [hunk 4]
```diff
@@ -1,10 +1,20 @@
...
+[![License](https://img.shields.io/:license-Apache%202-green.svg)](http://www.apache.org/licenses/LICENSE-2.0.txt)
 ![Continuous Integration](https://github.com/pitgull/pitgull/workflows/Continuous%20Integration/badge.svg)
 [![Powered by cats](https://img.shields.io/badge/powered%20by-cats-blue.svg)](https://github.com/typelevel/cats)
 ![Gluten free](https://img.shields.io/badge/gluten-free-orange.svg)
 
+## How it works
+
+The core idea behind the project is very simple - create a tool for automatically applying merge requests created by [Scala Steward](https://github.com/scala-steward-org/scala-steward) on any [Gitlab](https://gitlab.com/) instance. Currently Pitgull is ortogonal from the bots, the merge requests are qualified by generic rules.
+
+Pitgull works as a web service, listening for webhook events from Gitlab instance. Upon receiving a webhook, it reads related project's merge requests, finds one best candidate and tries to merge it. If all the qualifying merge requests need to be updated, an attempt to rebase them onto the target branch will be made.
```
Comment (pkowalcze) [inline README.md:12]:
```suggestion
Pitgull works as a web service, listening for webhook events from Gitlab instance. Upon receiving a webhook, it reads related project's merge requests, finds one best candidate and tries to merge it. If all qualifying merge requests need to be updated, an attempt to rebase them onto the target branch will be made.
```

**Review by pkowalcze — APPROVED** [review]

**Pull request merged**